### ISSUE TITLE
Outliner - Add a user toggle to turn collection color block highlights on or off

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#4788]
https://github.com/user-attachments/assets/dc6e5828-cba2-4e90-847e-66d24fbb616e

We can now have color highlights over collections, it's a hot requested feature and thanks to Pablo Vasquez, we could implement it a bit easier. 

Now, some users may not like it, we are about customizability, so it would be nice to have a property in the filter drop down of the outliner to turn these off or on.

**Comment by Blenux** [comment]
Tried to do this, but have no idea how bah

**Comment by Draise14** [comment]
Yeah...

1. You gotta make a new C property, which is.. I don't know where
2. Once that is done, it needs to be registered for compile
3. Then in the operator drawing code, it needs to be called
4. Then this bool needs to be registered to the python api area, not sure where
5. Then you make the gui toggle. 

I think that's the theory. There are plenty of user prop conditionals everywhere, I just haven't back tracked far enough to know how they are built.

**Comment by ReinerBforartists** [comment]
You can have a look how other toggles are implemented in C. The toolshelf tabs toggle for example was implemented by us. I'll have a look if i can find the task

![toolshelftabs](https://github.com/user-attachments/assets/c8ec2be2-a3a3-4a2d-a25e-f0d14fd6651c)

**Comment by ReinerBforartists** [comment]
Aww, the rename from lotsa files from c to cc makes it impossible to find the commit. The log ends at the rename date. But you can simply follow the rna_def_property and the comment down the rabbit hole. Starts in source\blender\makesrna\intern\rna_space.cc

![rnaspacecc](https://github.com/user-attachments/assets/d5869841-ecb3-4b9c-ace8-662657ea07cc)

**Comment by ReinerBforartists** [comment]
Oh, just noticed that we need to change the NULL to nullptr. NULL is the old C way. Will make it a task.

**Comment by Blenux** [comment]
Ok will have another good today hopefully.

**Comment by ReinerBforartists** [comment]
How's it going?

**Comment by Blenux** [comment]
Yeah have no idea sadly :(

**Comment by ReinerBforartists** [comment]
Shall i assist you? It's as easy as duplicating the rna_def_property, adding a bitmask, then adding the menu entry and an if to the code for the colored outliner parts :)

**Comment by Blenux** [comment]
Sure having a huge brain fog atm bah

**Comment by ReinerBforartists** [comment]
Mh, i have assigned myself to another task yesterday already. Lemme first finish this one, and then i'll try to tackle this one here with you.

**Comment by ReinerBforartists** [comment]
Yeah, we need to follow one of the props in the preferences. The state of these props are stored automatically. The state of the one from gridflag not. These are stored with saving the startup.blend

**Comment by ReinerBforartists** [comment]
Now to this sweetie here. First, where is the commit for the color block hilighting. We need it to put it into the if.

**Comment by ReinerBforartists** [comment]
Here: #4787

Now let's check if we have anything outliner related in the prefs.

**Comment by ReinerBforartists** [comment]
Hmm, nope. Closest is editing tab. So let's make a outliner panel there, analyze and duplicate one of these checkboxes, maybe the one from the 3d cursor or the node editor. And then let's have a look if we can get it to work.

I'll make this task a branch now. That's a bigger operation, which may fail.

![left](https://github.com/user-attachments/assets/d0a96e41-fcaa-4ffc-9140-a48945edb2a5)

**Comment by ReinerBforartists** [comment]
We have already an BFA implemented prop here. So let's follow this one down the rabbit hole.

And we have a comment here that may or may not be valid. Such comments are of no use ...

**Comment by ReinerBforartists** [comment]
![newtab](https://github.com/user-attachments/assets/c41edbab-92b0-4b91-80da-13cb54d85044)

**Comment by ReinerBforartists** [comment]
Hmm, makes things much more complicated than i wanted it to be. But let's have a look if we can create the toggle.

**Comment by ReinerBforartists** [comment]
Okay, that was the flag itself. Three new entries. No crash, means i hopefully did everything right. And it stores its state automatically when turning off or on since it is preferences now. 

Not sure where to add it in the DNA_userdef_types.h file so far. There is no explicit enum for the outliner yet. And i have no idea how this is all connected, so i added it to the sequencer flags for now.  It might be that it is not functional and just works in the sequencer at the moment. But this is an easy fix then. Just wanted to add the rough functionality so that i can toy around.

Now comes the moment of truth, adding this flag to the colored collection rows c code ...

@Blenux, do you want to take over here? Hard work is done :P

![flagimplemented](https://github.com/user-attachments/assets/6e06f090-60a7-46cd-a337-ec27ef887bb9)

**Comment by Blenux** [comment]
I think I got it, though so bah I couldn't work it out myself grrr lol

**Comment by Blenux** [comment]
https://github.com/user-attachments/assets/c2d36ddb-f089-4600-a55f-bdd8c9b1d55c

**Comment by Blenux** [comment]
Hehe cheers, thought might be best to add a Outliner Userdef flag instead of using the Sequencer Editor one?

![image](https://github.com/user-attachments/assets/6eca9384-d8d9-4895-8960-458231cba6fa)

**Comment by ReinerBforartists** [comment]
Definitely :)
